service/Microsoft.Azure.Devices/ServiceAuthenticationWithDeviceSharedAccessPolicyToken.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Azure.Devices
+{
+    public class ServiceAuthenticationWithDeviceSharedAccessPolicyToken : IAuthenticationMethod
```

Comment:
Nit: Public classes must have documentation - this would add yet a new warning in the build system.